test/integration/isolated/skip.go

```diff
@@ -0,0 +1,25 @@
+//go:build integration_tests
+
+package isolated
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/testenv"
+)
+
+func SkipIfRouterNotExpressions(ctx context.Context, t *testing.T, _ *envconf.Config) context.Context {
+	flavor := testenv.KongRouterFlavor()
+	// TODO refactor the router flavor consts.
```

Comment:
Why not reuse https://github.com/Kong/kubernetes-ingress-controller/blob/4906c9a6c123f1aea4e10a53fe2118659a29a11c/test/integration/skip_helpers_test.go#L19-L27

Reply:
Good point but in here we're relying on setting the router flavor via env variables to prevent deploying kong into cluster (or rather performing full test setup) if the test has unmatched router flavor criteria.

e.g

https://github.com/Kong/kubernetes-ingress-controller/blob/251ae9b42b5b2bce4c3b1680ee96cbd32e9887ff/test/integration/isolated/udproute_test.go#L488-L489